File: src/ccasp/wizard.js
import inquirer from 'inquirer';
import { detectOS, detectRuntimes } from './detect.js';
import { runInit, scanProject } from './init.js';

export async function runWizard(options = {}) {
  const root = process.cwd();
  const osInfo = detectOS();
  const runtimeInfo = detectRuntimes();
  const status = scanProject(root);

  console.log('Status Summary\n=');
  console.log(`Project: ${status.root}`);
  console.log(`.ccasp/ present: ${status.hasCcasp ? 'yes' : 'no'}`);
  console.log(`.claude present: ${status.hasClaude ? 'yes' : 'no'}`);
  console.log(`.codex present: ${status.hasCodex ? 'yes' : 'no'}`);
  console.log(`CLAUDE.md present: ${status.hasClaudeMd ? 'yes' : 'no'}`);
  console.log(`Legacy commands: ${status.hasLegacyCommands ? 'yes' : 'no'}`);
  console.log(`Legacy hooks: ${status.hasLegacyHooks ? 'yes' : 'no'}`);
  if (status.brokenClaude) console.log('Broken .claude link detected');
  if (status.brokenCodex) console.log('Broken .codex link detected');
  console.log('');

  console.log(`OS: ${osInfo.platform}${osInfo.isWSL ? ' (WSL)' : ''}`);
  console.log(`Claude CLI: ${runtimeInfo.claude ? 'yes' : 'no'}`);
  console.log(`Codex CLI: ${runtimeInfo.codex ? 'yes' : 'no'}`);
  console.log('');

  const { pathChoice } = await inquirer.prompt([
    {
      type: 'list',
      name: 'pathChoice',
      message: 'Choose a setup path:',
      choices: [
        { name: 'Fresh setup', value: 'fresh' },
        { name: 'Convert legacy Claude project', value: 'legacy_claude' },
        { name: 'Convert legacy Codex project', value: 'legacy_codex' },
        { name: 'Merge dual runtime', value: 'merge' },
        { name: 'Repair broken symlinks', value: 'repair' },
      ],
    },
  ]);

  const { enableClaude, enableCodex } = await inquirer.prompt([
    { type: 'confirm', name: 'enableClaude', message: 'Enable Claude runtime?', default: true },
    { type: 'confirm', name: 'enableCodex', message: 'Enable Codex runtime?', default: runtimeInfo.codex },
  ]);

  const dualDefault = Boolean(enableClaude && enableCodex);
  const advanced = await inquirer.prompt([
    { type: 'confirm', name: 'dualRuntime', message: 'Enable dual runtime mode?', default: dualDefault },
    { type: 'confirm', name: 'hooksEnabled', message: 'Enable hooks?', default: true },
    {
      type: 'list',
      name: 'governance',
      message: 'Governance mode:',
      choices: [
        { name: 'Standard', value: 'standard' },
        { name: 'Strict enforcement', value: 'strict' },
        { name: 'Off', value: 'off' },
      ],
      default: 'standard',
    },
  ]);

  let defaultRuntime = enableClaude ? 'claude' : 'codex';
  if (enableClaude && enableCodex && advanced.dualRuntime) {
    const runtimeChoice = await inquirer.prompt([
      {
        type: 'list',
        name: 'defaultRuntime',
        message: 'Default runtime:',
        choices: [
          { name: 'Claude', value: 'claude' },
          { name: 'Codex', value: 'codex' },
        ],
        default: 'claude',
      },
    ]);
    defaultRuntime = runtimeChoice.defaultRuntime;
  }

  const convertLegacy = pathChoice === 'legacy_claude' || pathChoice === 'legacy_codex' || pathChoice === 'merge';
  const repair = pathChoice === 'repair' || options.repairLinks;

  console.log('Summary\n=');
  console.log(`Convert legacy: ${convertLegacy ? 'yes' : 'no'}`);
  console.log(`Repair links: ${repair ? 'yes' : 'no'}`);
  console.log(`Enable Claude: ${enableClaude ? 'yes' : 'no'}`);
  console.log(`Enable Codex: ${enableCodex ? 'yes' : 'no'}`);
  console.log(`Dual runtime: ${advanced.dualRuntime ? 'yes' : 'no'}`);
  console.log(`Hooks: ${advanced.hooksEnabled ? 'yes' : 'no'}`);
  console.log(`Governance: ${advanced.governance}`);
  console.log(`Default runtime: ${defaultRuntime}`);
  console.log('');

  const { confirm } = await inquirer.prompt([
    { type: 'confirm', name: 'confirm', message: 'Apply changes?', default: true },
  ]);

  if (!confirm) {
    console.log('No changes applied');
    return 0;
  }

  return runInit({
    dryRun: options.dryRun,
    repairLinks: repair,
    enableClaude,
    enableCodex,
    convertLegacy,
    configUpdates: {
      default_runtime: defaultRuntime,
      dual_runtime: advanced.dualRuntime,
      hooks_enabled: advanced.hooksEnabled,
      governance_mode: advanced.governance,
    },
  });
}

File: src/ccasp_cli/wizard.py
from dataclasses import dataclass
from typing import List

from .env import OSDetection, RuntimeDetection
from .project import InitPlan, ProjectStatus


@dataclass
class WizardChoice:
    key: str
    label: str


def choose(prompt: str, choices: List[WizardChoice], default_key: str) -> str:
    print(prompt)
    for choice in choices:
        marker = "*" if choice.key == default_key else " "
        print(f"  {marker} [{choice.key}] {choice.label}")
    while True:
        value = input(f"Select [{default_key}]: ").strip().lower() or default_key
        if any(c.key == value for c in choices):
            return value
        print("Invalid selection.")


def yes_no(prompt: str, default: bool) -> bool:
    suffix = "Y/n" if default else "y/N"
    while True:
        value = input(f"{prompt} [{suffix}]: ").strip().lower()
        if not value:
            return default
        if value in {"y", "yes"}:
            return True
        if value in {"n", "no"}:
            return False
        print("Please enter y or n.")


def plan_from_wizard(status: ProjectStatus, os_info: OSDetection, runtime_info: RuntimeDetection) -> InitPlan:
    print("CCASP Wizard")
    print("=")
    print(f"OS: {os_info.system}{' (WSL)' if os_info.is_wsl else ''}")
    print(f"Claude CLI: {'yes' if runtime_info.claude_cli else 'no'}")
    print(f"Codex CLI: {'yes' if runtime_info.codex_cli else 'no'}")
    print("")

    choices = [
        WizardChoice("fresh", "Fresh setup"),
        WizardChoice("legacy_claude", "Convert legacy Claude project"),
        WizardChoice("legacy_codex", "Convert legacy Codex project"),
        WizardChoice("merge", "Merge dual runtime"),
        WizardChoice("repair", "Repair broken symlinks"),
    ]
    selection = choose("Choose a setup path:", choices, default_key="fresh")

    convert_legacy = selection in {"legacy_claude", "legacy_codex", "merge"}
    repair_links = selection == "repair"

    enable_claude = yes_no("Enable Claude runtime?", default=True)
    enable_codex = yes_no("Enable Codex runtime?", default=False)

    if selection == "legacy_codex":
        enable_codex = True
    if selection == "legacy_claude":
        enable_claude = True

    print("")
    dual_runtime = yes_no("Enable dual runtime mode?", default=enable_claude and enable_codex)
    hooks_enabled = yes_no("Enable hooks?", default=True)
    governance = choose(
        "Governance mode:",
        [
            WizardChoice("standard", "Standard"),
            WizardChoice("strict", "Strict enforcement"),
            WizardChoice("off", "Off"),
        ],
        default_key="standard",
    )

    default_runtime = "claude" if enable_claude else "codex"
    if enable_claude and enable_codex and dual_runtime:
        default_runtime = choose(
            "Default runtime:",
            [WizardChoice("claude", "Claude"), WizardChoice("codex", "Codex")],
            default_key="claude",
        )

    print("")
    print("Summary:")
    print(f"  Convert legacy: {'yes' if convert_legacy else 'no'}")
    print(f"  Repair links: {'yes' if repair_links else 'no'}")
    print(f"  Enable Claude: {'yes' if enable_claude else 'no'}")
    print(f"  Enable Codex: {'yes' if enable_codex else 'no'}")
    print(f"  Dual runtime: {'yes' if dual_runtime else 'no'}")
    print(f"  Hooks: {'yes' if hooks_enabled else 'no'}")
    print(f"  Governance: {governance}")
    print(f"  Default runtime: {default_runtime}")
    print("")
    confirm = yes_no("Apply changes?", default=True)
    if not confirm:
        return InitPlan(enable_claude=False, enable_codex=False, convert_legacy=False, repair_links=False)

    return InitPlan(
        enable_claude=enable_claude,
        enable_codex=enable_codex,
        convert_legacy=convert_legacy,
        repair_links=repair_links,
        config_updates={
            "default_runtime": default_runtime,
            "dual_runtime": str(dual_runtime).lower(),
            "hooks_enabled": str(hooks_enabled).lower(),
            "governance_mode": governance,
        },
    )
